examples_tests/__init__.py

```diff
@@ -0,0 +1,161 @@
+    return candidates[0]
+
+
+class ExampleTestCase(testtools.TestCase):
+
+    example_dir = None
+
+    def setUp(self):
+        filter_ = config.get('filter', None)
+        if filter_:
+            if not re.match(filter_, self.example_dir):
+                self.skipTest(
+                    '{} does not match the filter {}'.format(
+                        self.example_dir, filter_))
+
+        super().setUp()
+        # To measure coverage, a wrapper for the snapcraft binary might be set
+        # in the environment variable.
+        snapcraft_bin = os.getenv('SNAPCRAFT', 'snapcraft')
+        self.snapcraft_command = os.path.join(
+            os.getcwd(), 'bin', snapcraft_bin)
+
+        self.useFixture(
+            fixtures.EnvironmentVariable('SNAPCRAFT_SETUP_PROXIES', '1'))
+
+        self.snappy_testbed = None
+        if not config.get('skip-install', False):
+            ip = config.get('ip', None)
+            if not ip:
+                self.snappy_testbed = self._set_up_qemu_testbed()
+            else:
+                port = config.get('port', None) or '22'
+                self.snappy_testbed = testbed.SshTestbed(
+                    ip, port, 'ubuntu')
+            self.snappy_testbed.wait()
+
+    def _set_up_qemu_testbed(self):
+        private_key = _get_latest_ssh_private_key()
+        snappy_image = config.get('snappy_image', None)
+        if not snappy_image:
+            temp_dir = tempfile.mkdtemp()
+            snappy_image = testbed.create_snappy_image(temp_dir)
+            # Store the image path in the config so it's only create once
+            # per execution.
+            config['snappy_image'] = snappy_image
+            # Delete the image when the execution exits.
+            atexit.register(shutil.rmtree, temp_dir)
+        snappy_testbed = testbed.QemuTestbed(
+            snappy_image, '8022', 'ubuntu', private_key)
+        snappy_testbed.create()
+        self.addCleanup(self.snappy_testbed.delete)
+        return snappy_testbed
+
+    def build_snap(self, example_dir):
+        working_dir = os.path.join('examples', example_dir)
+        subprocess.check_call(
+            [self.snapcraft_command, 'clean'], cwd=working_dir)
+        try:
+            subprocess.check_output(
+                [self.snapcraft_command, 'snap'], cwd=working_dir,
+                stderr=subprocess.STDOUT).decode('utf-8')
+        except subprocess.CalledProcessError as e:
+            self.addDetail('output', content.text_content(str(e.output)))
+            raise
+
+    def install_snap(self, example_dir, snap_name, version):
+        if not config.get('skip-install', False):
+            snap_file_name = '{}_{}_amd64.snap'.format(
+                snap_name, version)
+            snap_local_path = os.path.join(
+                'examples', example_dir, snap_file_name)
+            self.snappy_testbed.copy_file(snap_local_path, '/home/ubuntu')
+            self.snappy_testbed.run_command([
```

Comment:
I pushed a check for the output. Any possible exception will be raised and the test will fail, which I think is enough.